extensions/kubernetes/spi/src/main/java/io/quarkus/kubernetes/spi/KubernetesRoleBuildItem.java

```diff
@@ -1,16 +1,64 @@
 package io.quarkus.kubernetes.spi;
 
+import java.util.List;
+
 import io.quarkus.builder.item.MultiBuildItem;
 
 public final class KubernetesRoleBuildItem extends MultiBuildItem {
+    private final String name; // name of the Role resource
+    private final List<PolicyRule> rules;
+
+    public KubernetesRoleBuildItem(String name, List<PolicyRule> rules) {
+        this.name = name;
+        this.rules = rules;
+    }
 
-    private final String role;
+    public String getName() {
+        return name;
+    }
 
-    public KubernetesRoleBuildItem(String role) {
-        this.role = role;
+    public List<PolicyRule> getRules() {
+        return rules;
     }
 
-    public String getRole() {
-        return this.role;
+    public static final class PolicyRule {
+        private final List<String> apiGroups;
```

Comment:
I have no idea, this just mirrors the Kubernetes API :-)

API graduation (API group named "alpha", "beta", and final without designation) is certainly one case where this can be used. Another is older OpenShift resources that were in the `""` API group and are now moved to their own API group (run `oc get clusterrole view -o yaml` and see).

Reply:
Gotcha, thanks